src/coreclr/tools/aot/ILCompiler.MetadataTransform/Internal/Metadata/NativeFormat/Writer/NativeMetadataWriter.cs

```diff
@@ -1032,56 +977,16 @@ public static bool SequenceEqual<T>(this List<T> first, List<T> second)
 
         public static bool SequenceEqual<T>(this List<T> first, List<T> second, IEqualityComparer<T> comparer)
         {
-            if (first.Count != second.Count)
-            {
-                return false;
-            }
-
-            comparer ??= EqualityComparer<T>.Default;
-
-            for (int i = 0; i < first.Count; i++)
-            {
-                if (!comparer.Equals(first[i], second[i]))
-                {
-                    return false;
-                }
-            }
-
-            return true;
-        }
-
-        public static bool SequenceEqual<T>(this T[] first, T[] second)
-        {
-            return first.SequenceEqual(second, null);
-        }
-
-        public static bool SequenceEqual<T>(this T[] first, T[] second, IEqualityComparer<T> comparer)
-        {
-            if (first.Length != second.Length)
-            {
-                return false;
-            }
-
-            comparer ??= EqualityComparer<T>.Default;
-
-            for (int i = 0; i < first.Length; i++)
-            {
-                if (!comparer.Equals(first[i], second[i]))
-                {
-                    return false;
-                }
-            }
-
-            return true;
+            return CollectionsMarshal.AsSpan(first).SequenceEqual(CollectionsMarshal.AsSpan(second), comparer);
         }
     }
 
     // Distinguishes positive and negative zeros for float and double values
```

Comment:
This can be deleted, and the comparisons inlined into SingleComparer/DoubleComparer methods.

Reply:
ok